eng/testing/tests.mobile.targets

```diff
@@ -101,11 +117,36 @@
         <Output TaskParameter="AppBundlePath" PropertyName="AppBundlePath" />
         <Output TaskParameter="XcodeProjectPath" PropertyName="XcodeProjectPath" />
     </AppleAppBuilderTask>
+
+    <ItemGroup>
+      <_appFiles Include="$(AppBundlePath)/../**/*" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(_appFiles)"
+          DestinationFolder="$(BundleDir)/bin/%(RecursiveDir)" SkipUnchangedFiles="true"/>
```

Comment:
I think we should try and keep the minimal amount of duplicated files. I think we can handle this better by building the bundles and then copying just the bundle and the .sh scripts and zipping that up. This will have more files than needed in order to send to helix.

Reply:
The reason why this exists is due to us generating an .xcodeproj and building it, which has its own place to generate the .app.  That's not exactly easy to override, so we copy the app to `$(BundleDir)/bin`

Reply:
Yeah, I get the intention of this and it makes sense. However I mean we should just copy what we need, we're copying everything:
<_appFiles Include="$(AppBundlePath)/../**/*" />

We should just neede the .app and the .sh file, right? 

I think we should do this as part of `ArchiveTests` before zipping up, as this is only relevant for sending to helix.

Reply:
The app is a directory. That's why it seems more elaborate.

Reply:
We should just pass the list of `.app` folders to `<XHarnessiOSProject>` in the Helix SDK. Nothing else should be needed, no `.sh` no zipping, no result collection

Reply:
While I agree that is simple enough, we might move off from building the .app folders as part of the build and at some point move to build them into helix before running the tests. The reasoning behind that is since we have around 250 test projects, building all those bundles takes like 25-30 mins which is not ideal, and also we’ve hit issues with hard disk size, etc.

I think helix SDK should have an extensibility point to have xharness installed in the machine for custom work items. I don’t think everything should be abstracted as it tights our hands on optimizations like this.

Reply:
Sure, we can have this capability if you want to have more control over it manually.
Originally, there was also a plan that xharness would also do the packaging but this responsibility was then taken care of in `dotnet/runtime`. I think it might happen we move this to xharness later to enable it for other repositories too.

So my question is - what do we do right now? Since it seems we are building app bundles here for now, should we just use the Helix SDK the way I described above and we can talk about a new SDK functionality that would just prepare the XHarness command for the job and nothing else?

My only concern here is that we shouldn't leave it like it is now because the `.sh` payload would have to be changed to look like the one the Helix SDK is using otherwise stuff won't work. And if we do that, since we are still polishing and changing the SDK still (for example now we are rolling out Xcode 11.5), we would have to start maintaining the same code here as well.

Reply:
@safern , @steveisok - as @premun wrote already, using Helix SDK is the right thing to do as it will take care of the necessary steps and would solve many of the underlying issues. Duplicating the functionality in scripts doesn't seem right. Please let us know if you need some assistance with using the SDK or if you found that some functionality is missing. We'd be happy to discuss it with you and help resolving whatever is needed. Thanks!

Reply:
We should not build app during test runs as in the same configuration it won't be even supported. It's reasonable to expect that we build for example Android test apps and deploy it to emulators on 3 different operating systems (e.g. 3 different helix queues).

We also don't want to use XHarness for app building/publishing as that would cause pain with bootstrap dependencies without duplicating a lot of logic. Imagine we need a newer version of C# compiler to compile a test and we want to test that the C# compiler is generating right code for iOS with AOT codegen (that's just one example).

Reply:
I think there is a lot of confusion happening here. 

My understanding is:

1. Helix SDK will just setup the machine and invoke Xharness even if we point it to the .app bundle directory. I’m not suggesting we do anything different, I’m suggesting that we shouldn’t need to point Helix SDK to the .app bundle directory, but we can set a flag and the same setup will happen, but instead we will invoke xharness ourselves, at the end is the same behavior.

2. Building the tests will always happen on the build machine, I’m not suggesting compiling the tests on helix, I was suggesting just generating the bundle in the helix machine, which builds the xcodeproj and it is not that paralyzed as building managed code. I built a regular test build in my computer and takes around 3-5 mins. Building all tests for iOS in the same machine sequentially takes 20-25 mins. From my understanding once we published the app the C# compiler doesn’t come into play anymore and we’re just building native assets and generating the .app bundle.

Cc: @EgorBo @akoeplinger @steveisok

Reply:
@premun @safern @directhex I'll set up some time this week and we can get on the same page.  There are enough subtleties that can make things confusing.

Reply:
Yes, there is some misunderstanding. I thought it might have been cleared up already since you seem to be ahead in running Android and iOS and Android Helix SDK support stories are exactly the same. I just didn't realize people don't use the Helix SDK.

@safern
1. The Helix SDK was extended with XHarness targets that I was mentioning earlier that accept a task item list of .app folder paths and does everything for you (zips them, adds the job script to the payload, sends it to helix, collects results..).